app/v2/doc.go

```diff
@@ -0,0 +1,101 @@
+/*
+Package app provides an application framework that manages the live cycle of a running application.
+
+An app could be an HTTP server (or any kind of server), a worker or a simple program. Independently of the kind of the app, it always exposes an admin port at 9000 by default which serves metrics, debug information and kubernetes probes. It also is capable of graceful shutdown on receiving signals of terminating by itself.
+
+The recommended way to use the `app` package is to rely on the 'default app'. The 'default app' is a global app that can be accessed by public functions on the app package (the app package can be seen as the application)
+
+There is a running example on `app/examples/servefiles`
+
+# Basics
+
+The first thing you should do bootstrap the application. This will initialize the configuration, reading it from the commandline or environment, initialize zerolog (based on the Config) and initialize the default global app.
+
+	app.Bootstrap(version, &cfg)
+
+After the bootstrap, the app will already be exposing the admin port and the readiness probe will be returning error, indicating that the application is not yet ready to receive requests. But the liveless probe will be returning success, indicating the app is alive.
```

Comment:
Nit: Maybe "and the readiness probe will be returning error" -> "and the readiness probe will initially be returning and error". I think this might reduce chances of someone being confused